build/common.go

```diff
@@ -0,0 +1,281 @@
+package build
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+var exname string = ""
+
+func getExecutableName(os string, arch string) string {
+	if exname == "" {
+		var err error
+		exname, err = getExecutableFromPluginJSON()
+		if err != nil {
+			exname = "set_exe_name_in_plugin_json" // warning in the final name?
```

Comment:
Why can't we return an error here instead?

Reply:
Maybe it could be... but lets do that in a follow on PR.  right now this is used in a few places that don't bubble errors